### PR TITLE
Chore: Migrate SecretsKVStore to dskit service

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This wraps SecretKVService store initialization in a dskit service.

**Why do we need this feature?**

Needed by https://github.com/grafana/grafana/pull/72123

Part of https://github.com/grafana/grafana-enterprise/issues/5550

**Who is this feature for?**

@grafana/grafana-app-platform-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/secrets/kvstore/service.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil || secretsPlugin == nil {
+		s.log.Error("failed to start remote secrets management plugin")
+		if isFatal, readErr := IsPluginStartupErrorFatal(ctx, namespacedKVStore); isFatal || readErr != nil {
+			// plugin error was fatal or there was an error determining if the error was fatal
+			s.log.Error("secrets management plugin is required to start -- exiting app")
+			if readErr != nil {
+				return nil, readErr
+			}
+			return nil, err
+		}
```
Comment (DanCech) [inline pkg/services/secrets/kvstore/service.go:163]: This whole thing seems very brittle vs the returned error indicating whether it's fatal or not, but it's also a direct copy of the existing code so...

It does appear to be the only reason this service depends on the kvstore service, so if we could tidy it up we could remove that dependency.

Comment (toddtreece) [reply pkg/services/secrets/kvstore/service.go]: > This whole thing seems very brittle vs the returned error indicating whether it's fatal or not

@DanCech yeah, agreed. it's pretty hard to follow, but i'm not sure i have enough context about what is going here to improve this without breaking things. any suggestions?

> It does appear to be the only reason this service depends on the kvstore service

the namespaced kvstore is also used at the bottom of this function

Comment (toddtreece) [reply pkg/services/secrets/kvstore/service.go]: @mmandrus i see some code comments added in #52552 related to removing pieces [in grafana 10](https://github.com/grafana/grafana/blob/b7ed2b02f309874afe33f147f61bf7090febfc46/pkg/services/secrets/kvstore/remote_plugin.go#L149). do you happen to know if the error handling can be simplified here similar to what @DanCech is suggesting?

Comment (mmandrus) [reply pkg/services/secrets/kvstore/service.go]: @toddtreece the G10 part should be disregarded at this point, as this still hasn't launched on the AMG side and there are still a lot of secrets that need to be unified.

Truth be told, I wrote this code when I was very new to Grafana and Golang, so I would suggest we clean it up if it's in scope. I like Dan's idea, have `StartAndReturnPlugin()` assess if an error is fatal itself and return a wrapped static error if so. Let me know if you'd like me to take a look at it.

Comment (toddtreece) [reply pkg/services/secrets/kvstore/service.go]: @mmandrus 👍 thanks for the context. i'll give it a shot and reach out if i encounter any issues
